process/transactions/transactionDBBuilder.go

```diff
@@ -180,14 +178,27 @@ func (dtb *dbTransactionBuilder) addScrsReceiverToAlteredAccounts(
 		}
 
 		egldBalanceNotChanged := scr.Value == "" || scr.Value == "0"
-		esdtBalanceNotChanged := scr.EsdtValue == "" || scr.EsdtValue == "0"
+		esdtBalanceNotChanged := scr.EsdtTokenIdentifier == ""
 		if egldBalanceNotChanged && esdtBalanceNotChanged {
 			// the smart contract results that don't alter the balance of the receiver address should be ignored
 			continue
 		}
 		encodedReceiverAddress := scr.Receiver
+
+		var nftNonceSTR string
```

Comment:
hard to understand the logic & requirement here

Reply:
added this logic in a new function.